src/Features/Core/Portable/FindUsages/IRemoteFindUsagesService.cs

```diff
@@ -110,21 +108,25 @@ public ValueTask ReportProgressAsync(int current, int maximum)
         public ValueTask SetSearchTitleAsync(string title)
             => _context.SetSearchTitleAsync(title);
 
-        public ValueTask OnDefinitionFoundAsync(SerializableDefinitionItem definition)
+        public async ValueTask OnDefinitionFoundAsync(SerializableDefinitionItem definition)
         {
             var id = definition.Id;
-            var rehydrated = definition.Rehydrate(_solution);
+            var rehydrated = await definition.RehydrateAsync(_solution, _context.CancellationToken).ConfigureAwait(false);
 
             lock (_idToDefinition)
             {
                 _idToDefinition.Add(id, rehydrated);
             }
 
-            return _context.OnDefinitionFoundAsync(rehydrated);
+            await _context.OnDefinitionFoundAsync(rehydrated).ConfigureAwait(false);
         }
 
-        public ValueTask OnReferenceFoundAsync(SerializableSourceReferenceItem reference)
-            => _context.OnReferenceFoundAsync(reference.Rehydrate(_solution, GetDefinition(reference.DefinitionId)));
+        public async ValueTask OnReferenceFoundAsync(SerializableSourceReferenceItem reference)
+        {
+            var rehydrated = await reference.RehydrateAsync(_solution, GetDefinition(reference.DefinitionId), _context.CancellationToken).ConfigureAwait(false);
```

Comment:
It's not clear to me if _context.CancellationToken is usable here: this would mean that if the client has requested cancellation, that we'll throw here, but that may end up remoting the exception to the server which then breaks OOP.

Reply:
i would expect that to be ok as any callbacks must be cancellation safe as well.  if not, we have serious issues on our hands.  @tmat ?

Reply:
So while testing this I think I got an assert once, which was worrisome. My worry is the callback may be expecting a different token than this one -- basically the callback creates it's own linked token in some way, and we need to be throwing with that one.

Reply:
Had a quick idea: while we wait for @tmat's answer on this, I can just add a try/catch here and eat the cancellation exception: there's nothing the remote side could do with the cancelation exception anyways, and presumably it's having it's own cancellation process which will also cancel the work off.